src/overlays/actors/ovl_Obj_Kibako2/z_obj_kibako2.c

```diff
@@ -37,35 +39,207 @@ static InitChainEntry D_8098EE8C[] = {
+    }
+    return !((flag >= 0) && Actor_GetChestFlag(globalCtx, flag));
+}
+
+void ObjKibako2_Break(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    Vec3f* thisPos;
+    Vec3f pos;
+    Vec3f velocity;
+    s16 angle;
+    s32 i;
+
+    thisPos = &this->dyna.actor.world.pos;
+    for (i = 0, angle = 0; i < 0x10; i++, angle += 0x4E20) {
+        f32 sn = Math_SinS(angle);
+        f32 cs = Math_CosS(angle);
+        f32 tempRand;
+        s32 phi_s0;
+
+        tempRand = Rand_ZeroOne() * 30.0f;
+        pos.x = sn * tempRand;
+        pos.y = (Rand_ZeroOne() * 10.0f) + 2.0f;
+        pos.z = cs * tempRand;
+        velocity.x = pos.x * 0.2f;
+        velocity.y = (Rand_ZeroOne() * 10.0f) + 2.0f;
+        velocity.z = pos.z * 0.2f;
+        pos.x += thisPos->x;
+        pos.y += thisPos->y;
+        pos.z += thisPos->z;
+        tempRand = Rand_ZeroOne();
+        if (tempRand < 0.05f) {
+            phi_s0 = 0x60;
+        } else if (tempRand < 0.7f) {
+            phi_s0 = 0x40;
+        } else {
+            phi_s0 = 0x20;
+        }
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &pos, -200, phi_s0, 28, 2, 0, (Rand_ZeroOne() * 30.0f) + 5.0f,
+                            0, 0, 70, KAKERA_COLOR_NONE, OBJECT_KIBAKO2, D_06001040);
+    }
+    func_800BBFB0(globalCtx, thisPos, 90.0f, 6, 100, 160, 1);
+}
+
+
+void ObjKibako2_SpawnCollectible(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 collectible;
+
+    collectible = func_800A8150(this->dyna.actor.params & 0x3F);
+    if (collectible >= 0) {
+        Item_DropCollectible(globalCtx, &this->dyna.actor.world.pos, collectible | GET_KIBAKO2_COLLECTIBLE_ID(this));
+    }
+}
+
+void ObjKibako2_SpawnSkulltula(ObjKibako2* this, GlobalContext* globalCtx) {
+    s16 yRotation;
+    s32 actorSpawnParam;
+    Actor* skulltula;
+
+    if (ObjKibako2_ContainsSkulltula(this, globalCtx)) {
+        actorSpawnParam = KIBAKO2_SKULLTULA_SPAWN_PARAM(this);
+        yRotation = ((u32) Rand_Next() >> 0x11) + this->dyna.actor.yawTowardsPlayer + 0xC000;
+        skulltula = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_SW, this->dyna.actor.world.pos.x,
+                                this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, 0, yRotation, 0,
+                                actorSpawnParam);
+        if (skulltula != NULL) {
+            skulltula->parent = &this->dyna.actor;
+            skulltula->velocity.y = 13.0f;
+            skulltula->speedXZ = 0.0f;
+        }
+    }
+}
+
+void ObjKibako2_SpawnContents(ObjKibako2* this, GlobalContext* globalCtx) {
+    if (((this->dyna.actor.params >> 0xF) & 1) == 0) {
+        ObjKibako2_SpawnCollectible(this, globalCtx);
+    } else {
+        ObjKibako2_SpawnSkulltula(this, globalCtx);
+    }
+}
+
+void ObjKibako2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
+    s16 tempParams;
+    s32 sp24;
+
+    sp24 = (this->dyna.actor.params >> 0xF) & 1;
```

Comment:
This is used several times, and needs a macro and enum; the above function suggests its main use is what spawns when breaking.

Reply:
I believe I did handled it correctly, but please let me know if it doesn't look right.

Reply:
Yes, this looks good now.